component/src/test/java/org/wso2/extension/siddhi/gpl/execution/streamingml/clustering/clustree/ClusTreeStreamProcessorExtensionTest.java

```diff
@@ -384,64 +384,253 @@ public void testClusTree2D_13() throws Exception {
+                        "select closestCentroidCoordinate1, closestCentroidCoordinate2, x, y " +
+                        "insert into OutputStream;");
+        try {
+            SiddhiAppRuntime siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime(inputStream + query);
+        } catch (Exception e) {
+            logger.info("Error caught");
+            AssertJUnit.assertTrue(e instanceof SiddhiAppCreationException);
+            AssertJUnit.assertTrue(e.getCause().getMessage().contains("noOfEventsToRefreshMacroModel should be a " +
+                    "positive integer but found -20"));
+        }
+    }
+
+    @Test
+    public void testClusTree2D_21() throws Exception {
+        logger.info("ClusTreeStreamProcessorExtension Test - Test case to validate maxHeightOfTree " +
+                "to be above a minimum value to produce required noOfClusters");
+        SiddhiManager siddhiManager = new SiddhiManager();
+        String inputStream = "define stream InputStream (x double, y double);";
+
+        String query = (
+                "@info(name = 'query1') " +
+                        "from InputStream#streamingml:clusTree(20, 10, 20, 2, 50, x, y) " +
+                        "select closestCentroidCoordinate1, closestCentroidCoordinate2, x, y " +
+                        "insert into OutputStream;");
+        try {
+            SiddhiAppRuntime siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime(inputStream + query);
+        } catch (Exception e) {
+            logger.info("Error caught");
+            AssertJUnit.assertTrue(e instanceof SiddhiAppCreationException);
+            AssertJUnit.assertTrue(e.getCause().getMessage().contains("maxHeightOfTree should be an int greater " +
+                    "than 2.7268 but found 2"));
+        }
+    }
+
+    @Test
+    public void testClusTree2D_22() throws Exception {
+        logger.info("ClusTreeStreamProcessorExtension Test - Test case to validate horizon to be positive");
+        SiddhiManager siddhiManager = new SiddhiManager();
+        String inputStream = "define stream InputStream (x double, y double);";
+
+        String query = (
+                "@info(name = 'query1') " +
+                        "from InputStream#streamingml:clusTree(2, 10, 20, 2, -50, x, y) " +
+                        "select closestCentroidCoordinate1, closestCentroidCoordinate2, x, y " +
+                        "insert into OutputStream;");
+        try {
+            SiddhiAppRuntime siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime(inputStream + query);
+        } catch (Exception e) {
+            logger.info("Error caught");
+            AssertJUnit.assertTrue(e instanceof SiddhiAppCreationException);
+            AssertJUnit.assertTrue(e.getCause().getMessage().contains("horizon should be a positive integer but " +
+                    "found -50"));
         }
     }
+
+    /*@Test
+    public void testClusTree2D_23() throws Exception {
```

Comment:
Why this test case is commented?

Reply:
There was an issue with MOA library. Now bumped moa version and fixed in https://github.com/wso2-extensions/siddhi-gpl-execution-streamingml/pull/30